### ISSUE TITLE
model with best auc not necessarily saved

### ISSUE THREAD

**Issue by magomagic** [brucechou1983/CheXNet-Keras#33]
in some cases validation loss can be high yet with better mean auc.current code copies weights.h5 (which are monitored according to val_loss.) to best_weights which is a bug.best weights should be saved from model and not copied
